src/x86_64/pvclock.c

```diff
@@ -61,17 +54,24 @@ closure_function(0, 1, void, tsc_deadline_timer,
     write_msr(TSC_DEADLINE_MSR, rdtsc() + count);
 }
 
-clock_timer init_tsc_deadline_timer(void)
+closure_function(1, 0, void, tsc_deadline_percpu_init,
+                 int, irq)
+{
+    lapic_set_tsc_deadline_mode(bound(irq));
```

Comment:
Actually they don't, at least for lapic timers which are local to the cpu. runloop can happen on any cpu, and the next timer event might need to be reprogrammed. As you may recall, I thought about sending an IPI to the bsp just to have it set the timer there. But that's kind of silly, as all the cores have the same timer hardware. So I just have it set the timer on whatever cpu it happens to be running on. The only kind of gross thing is that a timer may still be set to expire on another cpu...and it will fire if runloop doesn't happen to run on that cpu and reprogram the timer beforehand.

It's a compromise...I figured a spurious timer wakeup was less offensive than doing an IPI just to set a timer on the bsp when one is available on the current cpu. Maybe there's a more elegant solution that I'm not seeing.

In the case of xen or hpet, it's just one system-wide timer, and this per cpu init is ignored.